### ISSUE TITLE
TotK still crashes on loading screen > Vulkan: Device loss occured! [3589]

### ISSUE THREAD

**Issue by jdhau** [yuzu-emu/yuzu#10290]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Affected Build(s)

EA3589

### Description of Issue

Whenever I try to start a game on TotK, Yuzu crashes. It seems to be a shader of sorts, as the log file just dies halfway in a sentence. Worth to note that BotW runs and plays fine and that the game runs fine in Ryujinx.

Of note are these lines in the log:

```
[  24.975526] Render.Vulkan <Info> video_core\vulkan_common\vulkan_device.cpp:GetSuitability:697: Device doesn't support extension VK_NV_geometry_shader_passthrough
[  24.975531] Render.Vulkan <Info> video_core\vulkan_common\vulkan_device.cpp:GetSuitability:697: Device doesn't support extension VK_NV_viewport_array2
[  24.975532] Render.Vulkan <Info> video_core\vulkan_common\vulkan_device.cpp:GetSuitability:697: Device doesn't support extension VK_NV_viewport_swizzle
[  25.113691] Render.Vulkan <Info> video_core\vulkan_common\vulkan_device.cpp:GetSuitability:697: Device doesn't support extension VK_NV_geometry_shader_passthrough
[  25.113696] Render.Vulkan <Info> video_core\vulkan_common\vulkan_device.cpp:GetSuitability:697: Device doesn't support extension VK_NV_viewport_array2
[  25.113697] Render.Vulkan <Info> video_core\vulkan_common\vulkan_device.cpp:GetSuitability:697: Device doesn't support extension VK_NV_viewport_swizzle
[  87.415655] Render.Vulkan <Critical> video_core\vulkan_common\vulkan_device.cpp:ReportLoss:527: Device loss occurred!
```

### Expected Behavior

Game doesn't crash.

### Reproduction Steps

Specs:

> AMD Ryzen 6800u
> 680M iGPU (2Gb VRAM) / Driver 23.4.3
> 16Gb DDR5-6400
> Latest AMD drivers

Clear shaders. Load game. Game crashes after loading screen.

### Log File

Full log [here](https://gist.github.com/jdhau/d001380dfe44eb4de376b1b154b67c69).

### System Configuration

CPU: Ryzen 6800u
GPU/Driver: Radeon 680M (RDNA2) / 23.4.3
RAM: 16Gb DDR5-6400
OS: Win 11

**Comment by gidoly** [comment]
Last amd driver is unstable with yuzu revert to 23.3.1

**Comment by Squall-Leonhart** [comment]
> Last amd driver is unstable with yuzu revert to 23.3.1

This was fixed with 23.4.2 and later.

**Comment by jdhau** [comment]
Just reverted. Same issue still. Tried 16.0.1 and 16.0.2 keys.

**Comment by jdhau** [comment]
I jsut got it to load into my save for the first time. It loaded but crashed on the first frame. Log:

```
[ 463.430322] Render.Vulkan <Info> video_core\renderer_vulkan\vk_pipeline_cache.cpp:CreateGraphicsPipeline:571: 0x0f2980d985396548
[ 463.458018] Render.Vulkan <Info> video_core\renderer_vulkan\vk_pipeline_cache.cpp:CreateGraphicsPipeline:571: 0x567e711c64d2702b
[ 464.077751] Render.Vulkan <Info> video_core\renderer_vulkan\vk_pipeline_cache.cpp:CreateGraphicsPipeline:571: 0xb7d1a1892c04c84b
[ 465.376872] Render.Vulkan <Info> video_core\renderer_vulkan\vk_pipeline_cache.cpp:CreateGraphicsPipeline:571: 0x8eecc6d99983b59b
[ 465.386046] Render.Vulkan <Info> video_core\renderer_vulkan\vk_pipeline_cache.cpp:CreateGraphicsPipeline:571: 0xb733bac27d51e029
[ 465.386980] Render.Vulkan <Info> video_core\renderer_vulkan\vk_pipeline_cache.cpp:CreateGraphicsPipeline:571: 0x3474117e6bd95fb1
[ 465.388248] Render.Vulkan <Info> video_core\renderer_vulkan\vk_pipeline_cache.cpp:CreateGraphicsPipeline:571: 0xc3abb5a80d8d3232
[ 465.613564] Render.Vulkan <Critical> video_core\vulkan_common\vulkan_device.cpp:ReportLoss:527: Device loss occurred!
[ 465.635085] Render.Vulkan <Info> video_core\renderer_vulkan\vk_pipeline_cache.cpp:CreateGraphicsPipeline:571: 0xf6632e21bb104539
[ 465.646877] Debug <Critical> video_core\renderer_vulkan\vk_staging_buffer_pool.cpp:operator ():159: Assertion Failed!
[ 465.646882] Debug <Critical> video_core\renderer_vulkan\vk_staging_buffer_pool.cpp:operator ():160: Assertion Failed!
[ 465.696897] Debug <Critical> video_core\renderer_vulkan\vk_staging_buffer_pool.cpp:operator ():159: Assertion Failed!
[ 465.696901] Debug <Critical> video_core\renderer_vulkan\vk_staging_buffer_pool.cpp:operator ():160: Assertion Failed!
[ 465.696903] Debug <Critical> video_core\renderer_vulkan\vk_staging_buffer_pool.cpp:operator ():159: Assertion Failed!
[ 465.696904] Debug <Critical> video_core\renderer_vulkan\vk_staging_buffer_pool.cpp:operator ():160: Assertion Failed!
```

**Comment by lorki0052** [comment]
[ 114.997023] Render.Vulkan <Info> video_core\renderer_vulkan\vk_pipeline_cache.cpp:CreateGraphicsPipeline:571: 0xc143caf36c0cf02e
[ 114.997222] Render.Vulkan <Info> video_core\renderer_vulkan\vk_pipeline_cache.cpp:CreateGraphicsPipeline:571: 0x4fbc2ebe308f23a4
[ 114.997964] Render.Vulkan <Info> video_core\renderer_vulkan\vk_pipeline_cache.cpp:CreateGraphicsPipeline:571: 0x30c06f602eccddf6
[ 116.522851] Render.Vulkan <Critical> video_core\vulkan_common\vulkan_device.cpp:ReportLoss:527: Device loss occurred!

Same here. 

> ### Specs:
> 
> Intel(R) Core(TM) i5-6500 CPU @ 3.20GHz   3.20 GHz
> 970 GTX/Driver 531.79
> 16Gb DDR3

Game crashes after loading screen "Nintendo Presents".

**Comment by Squall-Leonhart** [comment]
>  970 GTX/Driver 531.79

900 series needs "Accelerate ASTC Decoding" disabled on the graphics settings when using drivers newer than 472.12

**Comment by jdhau** [comment]
Seems like this issue is a bit larger than just TotK. Metroid Prime had something similar just now:

```
589.200180] Render.Vulkan <Info> video_core\renderer_vulkan\vk_pipeline_cache.cpp:CreateGraphicsPipeline:571: 0xe50f1572ce315b9e
[ 589.202588] Render.Vulkan <Info> video_core\renderer_vulkan\vk_pipeline_cache.cpp:CreateGraphicsPipeline:571: 0x2c03a2e1ae5dcd56
[ 589.368068] Render.Vulkan <Info> video_core\renderer_vulkan\vk_pipeline_cache.cpp:CreateGraphicsPipeline:571: 0xe9e1506a99bbfa32
[ 589.435067] Render.Vulkan <Info> video_core\renderer_vulkan\vk_pipeline_cache.cpp:CreateGraphicsPipeline:571: 0x2dd45589ba24a4ef
[ 589.448681] Render.Vulkan <Info> video_core\renderer_vulkan\vk_pipeline_cache.cpp:CreateGraphicsPipeline:571: 0x1016658bdbf13412
[ 589.449818] Render.Vulkan <Info> video_core\renderer_vulkan\vk_pipeline_cache.cpp:CreateGraphicsPipeline:571: 0x374229d201755efc
[ 589.450515] Render.Vulkan <Info> video_core\renderer_vulkan\vk_pipeline_cache.cpp:CreateGraphicsPipeline:571: 0x362a6ab9c4b34a00
[ 589.463087] Render.Vulkan <Info> video_core\renderer_vulkan\vk_pipeline_cache.cpp:CreateGraphicsPipeline:571: 0x1e791fd909153d43
[ 591.886284] Render.Vulkan <Error> video_core\renderer_vulkan\vk_swapchain.cpp:AcquireNextImage:150: vkAcquireNextImageKHR returned VK_ERROR_DEVICE_LOST
[ 591.886338] Render.Vulkan <Critical> video_core\vulkan_common\vulkan_device.cpp:ReportLoss:527: Device loss occurred!
[ 591.915886] Debug <Critical> video_core\renderer_vulkan\vk_staging_buffer_pool.cpp:operator ():159: Assertion Failed!
[ 591.915891] Debug <Critical> video_core\renderer_vulkan\vk_staging_buffer_pool.cpp:operator ():160: Assertion Failed!
```

**Comment by lorki0052** [comment]
> > 970 GTX/Driver 531.79
> 
> 900 series needs "Accelerate ASTC Decoding" disabled on the graphics settings when using drivers newer than 472.12

I checked it, any combination of checkboxes just didn't achieve results. Only Opengl works fine. It seems to be a problem with the drivers on the Yuzu side. The ryujinx doesn't have this problem.

**Comment by BenjaminHalko** [comment]
I can confirm this issue is also present on Nvidia GTX 1050 with driver 472.12.

**Comment by gitgud187** [comment]
Same thing happening to me aswell on Ryzen 6800u | 680m

**Comment by wlwetzel** [comment]
On previous builds totk would crash right after the ganon cutscene, on 3589 it crashes immediately. R9 380

Render.Vulkan <Critical> video_core\vulkan_common\vulkan_device.cpp:ReportLoss:527: Device loss occurred!

**Comment by Jaitsu** [comment]
I'm running into this too, and I'm curious. To those affected, are you on laptops with hybrid graphics? I'm on such a laptop (Asus G731GU, i7-9750H/mobile 1660 Ti) I'm still testing, but I'm wondering if it has to do with the GPU being suspended in the time between shaders loading and the game starting.

I just got this crash 4X in a row ("Device loss occurred!"), then tried running the Nvidia control panel on the "Adjust image settings with preview" page in order to keep the GPU active, and the game started 4/5 times after of that (1/5 got stuck on the black screen after "Launching..." but notably did NOT actually crash or produce the "Device loss occurred!" error)

**Comment by jay9ee** [comment]
I'm on a Lenovo Yoga 14ARB7 with the 6800u and Radeon 680m so no hybrid graphics. I'd imagine the 2 other posters with the same CPU don't have hybrid graphics either as that CPU is mostly for slim Notebooks.

**Comment by Monoverde888** [comment]
same thing happens here on an uhd 630 that previously ran the game decently

**Comment by jdhau** [comment]
> I'm on a Lenovo Yoga 14ARB7 with the 6800u and Radeon 680m so no hybrid graphics. I'd imagine the 2 other posters with the same CPU don't have hybrid graphics either as that CPU is mostly for slim Notebooks.

Yep. Same laptop.

**Comment by vasilvestre** [comment]
It also happened to me on Steam link, I removed ReShader and Vulkan pipeline cache.
Steam got his own Vulkan Pipeline cache so maybe there's a conflict ?

**Comment by TheShotte** [comment]
I'm having a similar issue. After about a minute or 2 of gameplay, yuzu just hangs and then CTDs. This is the end of the log:

[ 393.643196] Render.Vulkan <Info> video_core\renderer_vulkan\vk_pipeline_cache.cpp:CreateGraphicsPipeline:571: 0x59a953d7f7ba5497
[ 396.966138] Render.Vulkan <Critical> video_core\vulkan_common\vulkan_device.cpp:ReportLoss:527: Device loss occurred!

i7 12700K
AMD 6600XT 23.4.3
32 Gb RAM
NVMe SSD

This is on the EA 3591,

**Comment by alissongamer** [comment]
Same error here, I've done everything and the error persists.
GPU: AMD R9 290
CPU: i5 ninth generation
RAM: 8GB

The error doesn't occur if running in OpenGL, however the performance is very bad here and with graphical bugs. :(

**Comment by megaminoxer** [comment]
Idk if its the same issue for me but i recorded a [video](https://drive.google.com/file/d/1Xq6FJk18i0SvOliAOGdEFu6O_A2uegtg/view?usp=sharing ) showing the console while running.

Intel(R) Core(TM) i7-6700 CPU
AMD Radeon R9 390x GPU (yea very old)
16GB RAM 
EA 3591

Crashes immediately after the first cutscene.

[log](https://drive.google.com/file/d/1U5efTHMc8oPo5GNaqxAZEfwK1VBCUDlT/view?usp=share_link)
_________

!I own the game and just want to emulate it for testing purposes!

**Comment by JonathanJJohnson** [comment]
Same error for me, it seems it occurs when the game switches from cutscene to gameplay. It happened after the intro sequence
 and then again (after I load the save) when the Master Sword glows, a few steps down the cave. I'm a Arc User.

Intel Core i5-13600k
Intel Arc A750
32GB DDR5 RAM

[yuzu_log.txt](https://github.com/yuzu-emu/yuzu/files/11481810/yuzu_log.txt)

**Comment by JoRyJuKy** [comment]
I also recently crashed seemingly due to this issue, after a decently long playtime, just after clearing out a cave in ToTK. I'm running a RTX2070Super, Ryzen 7 5800x, and 32gb of ram

[yuzu_log.txt](https://github.com/yuzu-emu/yuzu/files/11482985/yuzu_log.txt)

**Comment by schbzz** [comment]
had the same issue in yuzu ea 3591, yuzu ea 3592 and pineappleEA 3593
ram: 16 gb
gpu: gtx 1060 3gb
cpu: ryzen 5 1600

<Critical>video_core\vulkan_common\vulkan_device.cpp:ReportLoss:527: Device loss occurred!

**Comment by Squall-Leonhart** [comment]
At this point, this looks to be a duplicate of #10274

**Comment by jdhau** [comment]
> At this point, this looks to be a duplicate of #10274

I'm not sure I agree. I can't even get to the weapon selection menu, as the game doesn't get beyond the in-game loading screen. Or are you implying that the underlying issue triggering this fault is the same?

**Comment by Squall-Leonhart** [comment]
> Or are you implying that the underlying issue triggering this fault is the same?

Yes, it all seems to be connected to video memory and the GC mechanism evicting data while its still needed, this is going to occur more so on systems where users have multiple hardware accelerated apps open in the tray, ie, discord, icue, LGS/Ghub, or open web browsers as these all add up and means the users video card has less available video memory at the time Yuzu starts.

In most cases vulkan treats a missing resource as a critical event and will throw a device lost crash, as it doesn't support hard pagefaults by design, (similiarly to D3D12) particularly if the texture is a component of a shader running or about to be run.

In other cases, like the event where a bitmap type texture, like that on a loading screen is evicted, it might softlock the application instead when it comes time to flip to it, this would be a source for the loading screens failing to transition on some devices.

**Comment by sysmso** [comment]
had the same issue with yuzu 3592 on windows 11
ram : 16G
gpu : GeForce GTX 970
cpu : AMD Ryzen 5 1600

but works fine on the same computer with Arch Linux and yuzu 3592

**Comment by jdhau** [comment]
> > Or are you implying that the underlying issue triggering this fault is the same?
> 
> Yes, it all seems to be connected to video memory and the GC mechanism evicting data while its still needed, this is going to occur more so on systems where users have multiple hardware accelerated apps open in the tray, ie, discord, icue, LGS/Ghub, or open web browsers as these all add up and means the users video card has less available video memory at the time Yuzu starts.
> 
> In most cases vulkan treats a missing resource as a critical event and will throw a device lost crash, as it doesn't support hard pagefaults by design, (similiarly to D3D12) particularly if the texture is a component of a shader running or about to be run.
> 
> In other cases, like the event where a bitmap type texture, like that on a loading screen is evicted, it might softlock the application instead when it comes time to flip to it, this would be a source for the loading screens failing to transition on some devices.

Right. Interesting. That does explain why I seem to have slightly different results (different amount of free VRAM) each time I run the game. Sometimes it crashes before it even reaches the loading screen, sometimes it loads into the game but crashes after a few seconds, etc. usually it bails after the loading screen though.

**Comment by Squall-Leonhart** [comment]
Liam has some idea's that he is looking into, I'm not sure i should mention them here in case some users start bombarding him with demands to do one or the other.

**Comment by snipes-woodland** [comment]
> > Or are you implying that the underlying issue triggering this fault is the same?
> 
> Yes, it all seems to be connected to video memory and the GC mechanism evicting data while its still needed, this is going to occur more so on systems where users have multiple hardware accelerated apps open in the tray, ie, discord, icue, LGS/Ghub, or open web browsers as these all add up and means the users video card has less available video memory at the time Yuzu starts.
> 
> In most cases vulkan treats a missing resource as a critical event and will throw a device lost crash, as it doesn't support hard pagefaults by design, (similiarly to D3D12) particularly if the texture is a component of a shader running or about to be run.
> 
> In other cases, like the event where a bitmap type texture, like that on a loading screen is evicted, it might softlock the application instead when it comes time to flip to it, this would be a source for the loading screens failing to transition on some devices.

I tested this by logging with the gpu driver and on my system the vram never even crosses half of it's capacity it seems to be the system ram for me that gets to 11,11 Gb then the crash seems to happen (12Gb ram)

**Comment by gitgud187** [comment]
This might not be surprising to the people who actually know what's going on but i just tried it and i'm not experiencing these issues on Linux, the game works absolutely fine there. On windows it still crashes. Yuzu EA 3594 - Ryzen 6800u Radeon 680m iGPU

**Comment by vasilvestre** [comment]
I played for hours, then I got the issue again today, I had to disable the cache again and it works now.

**Comment by Squall-Leonhart** [comment]
> I tested this by logging with the gpu driver and on my system the vram never even crosses half of it's capacity it seems to be the system ram for me that gets to 11,11 Gb then the crash seems to happen (12Gb ram)

Doesn't have to, video memory fragmentation is a thing that triggers OOM too.

**Comment by falz1994** [comment]
> This might not be surprising to the people who actually know what's going on but i just tried it and i'm not experiencing these issues on Linux, the game works absolutely fine there. On windows it still crashes. Yuzu EA 3594 - Ryzen 6800u Radeon 680m iGPU

Yep, same here. After saw Squall. comment about a feature being availeble in Linux and not in Windows, tried in Ubuntu and it works great, I only had do disable the option to Accelarate ATC Texture, or something like that. Is still a little laggy but way better than OPENGL

**Comment by alissongamer** [comment]
I already have the new Early Access version 3495 and the error still persists :(

**Comment by marikanoo** [comment]
Same here..game crashes on Vlucan during the save loading screen :(

**Comment by Bisc0tt1** [comment]
> This might not be surprising to the people who actually know what's going on but i just tried it and i'm not experiencing these issues on Linux, the game works absolutely fine there. On windows it still crashes. Yuzu EA 3594 - Ryzen 6800u Radeon 680m iGPU

I'm on Linux as well and have had two crashes in three hours of gameplay. They seem to be roughly similar to the crashes mentioned in this thread, because so far they've happened when exiting out of a menu or right when control is handed back to me after exiting a loading screen. Worst part is, the GPU driver just crashes completely and takes the entire system with it. I can't even spawn a TTY shell and am forced to hard reset the entire system, even after waiting for minutes for it to recover. Checking the kernel logs it's getting spammed with page fault messages like these when the system freeze happens:
```shell
amdgpu 0000:0b:00.0: amdgpu: [gfxhub] page fault (src_id:0 ring:24 vmid:4 pasid:32789, for process yuzu pid 2020 thread VulkanWorker pid 2144)
May 20 18:18:53 ambrosia kernel: amdgpu 0000:0b:00.0: amdgpu:   in page starting at address 0x000080031a4a5000 from client 0x1b (UTCL2)
May 20 18:18:53 ambrosia kernel: amdgpu 0000:0b:00.0: amdgpu: GCVM_L2_PROTECTION_FAULT_STATUS:0x00401031
May 20 18:18:53 ambrosia kernel: amdgpu 0000:0b:00.0: amdgpu:          Faulty UTCL2 client ID: TCP (0x8)
May 20 18:18:53 ambrosia kernel: amdgpu 0000:0b:00.0: amdgpu:          MORE_FAULTS: 0x1
May 20 18:18:53 ambrosia kernel: amdgpu 0000:0b:00.0: amdgpu:          WALKER_ERROR: 0x0
May 20 18:18:53 ambrosia kernel: amdgpu 0000:0b:00.0: amdgpu:          PERMISSION_FAULTS: 0x3
May 20 18:18:53 ambrosia kernel: amdgpu 0000:0b:00.0: amdgpu:          MAPPING_ERROR: 0x0
May 20 18:18:53 ambrosia kernel: amdgpu 0000:0b:00.0: amdgpu:          RW: 0x0
``` 

Running Yuzu EA 3599 with an RX6800 XT and Ryzen 9 5900X. It's definitely not all sunshine and rainbows on the Linux side either.

**Comment by JpauloHs** [comment]
After almost freaking out with this bug, changing the drive of the video card "GTX 1650", I did the simplest thing, I entered AMD's Vulkan® Driver Support and downloaded the automatic updater, rebooted the pc and it worked. I no speak very well so sorry for mistake.

**Comment by Jaitsu** [comment]
I've seen the above responses mentioning that they're on different laptops that don't have hybrid graphics - but I think there may be multiple causes for this issue.

I've been testing and testing and I _never_ get the "Device loss occurred!" issue if another application (like the Geforce Control Panel as I mentioned above) is running and keeping my hybrid-graphics laptop's dedicated GPU active during the loading process. If it's not running, it's a 50-50 shot. (I still get the black screen hang sometimes, but it's a much lower chance and does not produce the "Device loss occurred!" error.)

**Comment by Squall-Leonhart** [comment]
Once the driver has a binding on the Device, it doesn't go to sleep just because a moment of low utilisation occurs, the nvapi and powermiser api's keep the GPU awake, so your workaround is snake oil, sorry.

**Comment by Landsil** [comment]
Same here on SteamDeck on Win 11 but

1. Worked fine for 5min while I tested
2. Got Shaders from internet (I think it worked fine, not sure)
3. It's broken now
4. I removed all of the shaders I had
5. Works fine again
6. closed the game, re-added shaders
7. crash
8. Removed shaders again
9. **Works fine again**

**Comment by Merykantm** [comment]
> After almost freaking out with this bug, changing the drive of the video card "GTX 1650", I did the simplest thing, I entered AMD's Vulkan® Driver Support and downloaded the automatic updater, rebooted the pc and it worked. I no speak very well so sorry for mistake.

What is your Vulkan version now ?

**Comment by jdhau** [comment]
@Squall-Leonhart just proving a quick update in acse it is helpful.

I got the game to actually load and run a few times by constantly reloading the game. It seems to crash when it's building shaders before placing you in an area. Restarting over and over again and letting it build a couple of shaders each time got the game to run, but as soon as I tried fast traveling it crashed again. Using build 3601.

**Comment by ArrenLebannen** [comment]
same here but after i used EA3601 i was able to play it for 40 minutes without crashing and without using any mods or changing any settings, and after that i got the same issue with log device loss occurred, already reverted driver to 23.3.1 and disabling both ASTC decode still doesn't help

i3 10105f
RX 6500 xt 4gb / 23.4.3
16gb

maybe my poor pc is not strong enough to play this

**Comment by jdhau** [comment]
EA3605 doesn't even try to load a game. It just exits.

**Comment by Squall-Leonhart** [comment]
> EA3605 doesn't even try to load a game. It just exits.

Do a reinstall of the update data, someone else had this too and that fixed it. not sure why.

**Comment by ArrenLebannen** [comment]
> yeah that did the trick. Game still crashes at the end of the loading screen though. Same device loss error during shader build.

maybe we can just wait for amd to release the latest driver

**Comment by Squall-Leonhart** [comment]
> > > EA3605 doesn't even try to load a game. It just exits.
> > 
> > 
> > Do a reinstall of the update data, someone else had this too and that fixed it. not sure why.
> 
> yeah that did the trick. Game still crashes at the end of the loading screen though. Same device loss error during shader build. I can get the game to run by loading it a number of times until all the shaders fora scene are built. But then as soon as I fast travel or do anything else out of the ordinary, it crashes again.

did you try with 1.5x scaling, like i said in the other issue, its not a linear increase in memory to go from 1x to 2x, also try looking at the process gpu stats with Process Hacker, might surprise you how much vram goes up with resolution.

**Comment by Infinality** [comment]
> 
> Have you deleted the shaders between 1.1.0 and 1.1.1 updates?, have you shut down unnecessary applications? have you relaxed or reset your IMC settings? D3D12 and Vulkan are sensitive to unstable memory (And a lack of pagefile)

I have removed shaders regularly and I've tried right after boot into Windows with nothing else major open.  The pagefile is 36GB.  How would I adjust IMC settings?  (Not familiar with that)  Thank you.

**Comment by jdhau** [comment]
> did you try with 1.5x scaling, like i said in the other issue, its not a linear increase in memory to go from 1x to 2x, also try looking at the process gpu stats with Process Hacker, might surprise you how much vram goes up with resolution.

I've only ever tried 1x resolution.

**Comment by Squall-Leonhart** [comment]
> How would I adjust IMC settings? (Not familiar with that) Thank you.
These are set in the BIOS, i've rarely heard of issues with ryzen on auto memory configurations.

If you're confident in doing so, shut down your system and take out the graphics card and reinstall it into the slot.
Do the same with the PCIE 8/6 pin connectors and the ATX 12v 24pin on the motherboard, just to rule out any weird issues with the contacts having a resistance issue.

You really should at this point be able to boot the game fine.

> I've only ever tried 1x resolution. Jsut tried 0.75x and 0.5x and even that crashes.

Apologies, i've mixed up some names of posters, You're on the 2GB integrated RDNA apu (and are the OP).
Did you change the Transcoding option to BC1 in the advanced graphics settings?

**Comment by Infinality** [comment]
> These are set in the BIOS, i've rarely heard of issues with ryzen on auto memory configurations.
> 
> If you're confident in doing so, shut down your system and take out the graphics card and reinstall it into the slot. Do the same with the PCIE 8/6 pin connectors and the ATX 12v 24pin on the motherboard, just to rule out any weird issues with the contacts having a resistance issue.
> 
> You really should at this point be able to boot the game fine.

I've actually tried multiple graphics cards on the same system, including a 3060Ti, and multiple memory speeds (default and accelerated), and Vulkan still crashes consistently on TOTK and BOTW (but works fine on some other games).  I've also tried on older Motherboard firmware and latest firmware.  It's just a weird issue.

**Comment by Squall-Leonhart** [comment]
> I've actually tried multiple graphics cards on the same system, including a 3060Ti, and multiple memory speeds (default and accelerated), and Vulkan still crashes consistently on TOTK and BOTW (but works fine on some other games). I've also tried on older Motherboard firmware and latest firmware. It's just a weird issue.

Try this driver
https://developer.nvidia.com/downloads/vulkan-beta-53204-windows

**Comment by Infinality** [comment]
> Try this driver https://developer.nvidia.com/downloads/vulkan-beta-53204-windows

Same unfortunately.  One thing I notice is that there is always another "Critical" that accompanies the device loss, the first line below:

[  70.718219] Debug <Critical> core\hle\service\filesystem\fsp_srv.cpp:operator ():899: Assertion Failed!
[  70.731849] Service.FS <Warning> core\hle\service\filesystem\fsp_srv.cpp:OpenSaveDataInfoReaderOnlyCacheStorage:927: (STUBBED) called
[  70.755277] Core <Warning> core\hle\service\hle_ipc.cpp:WriteBuffer:361: skip empty buffer write
[  70.931323] Service.AM <Warning> core\hle\service\am\am.cpp:CreateCacheStorage:1760: (STUBBED) called with index=0, size=0x33b0000, journal_size=0x2c000
[  71.045883] Service.FS <Info> core\hle\service\filesystem\fsp_srv.cpp:OpenSaveDataFileSystem:875: called.
[  71.569711] Service.FS <Warning> core\hle\service\filesystem\fsp_srv.cpp:Commit:480: (STUBBED) called
[  71.809496] Service.NVDRV <Warning> core\hle\service\nvdrv\devices\nvhost_ctrl_gpu.cpp:ZBCSetTable:251: (STUBBED) called
[  71.809500] Service.NVDRV <Warning> core\hle\service\nvdrv\devices\nvhost_ctrl_gpu.cpp:ZBCSetTable:259: Avoiding passing null pointer to memcpy
[  71.809873] Service.NVDRV <Warning> core\hle\service\nvdrv\devices\nvhost_ctrl_gpu.cpp:ZBCSetTable:251: (STUBBED) called
[  71.809875] Service.NVDRV <Warning> core\hle\service\nvdrv\devices\nvhost_ctrl_gpu.cpp:ZBCSetTable:259: Avoiding passing null pointer to memcpy
[  72.227290] Service.FS <Warning> core\hle\service\filesystem\fsp_srv.cpp:Commit:480: (STUBBED) called
[  72.267989] Service.FS <Info> core\hle\service\filesystem\fsp_srv.cpp:OpenSaveDataFileSystem:875: called.
[  72.277480] Service.FS <Info> core\hle\service\filesystem\fsp_srv.cpp:OpenSaveDataFileSystem:875: called.
[  72.314179] Service.FS <Warning> core\hle\service\filesystem\fsp_srv.cpp:OpenReadOnlySaveDataFileSystem:911: (STUBBED) called, delegating to 51 OpenSaveDataFilesystem
[  72.314184] Service.FS <Info> core\hle\service\filesystem\fsp_srv.cpp:OpenSaveDataFileSystem:875: called.
[  81.616396] Service <Warning> core\hle\service\sm\sm_controller.cpp:QueryPointerBufferSize:72: (STUBBED) called
[  83.496101] Service <Warning> core\hle\service\sm\sm_controller.cpp:QueryPointerBufferSize:72: (STUBBED) called
[  83.503047] Service.NFC <Info> core\hle\service\nfc\nfc_interface.cpp:Initialize:28: called
[  83.504718] Service.NFC <Info> core\hle\service\nfc\nfc_interface.cpp:AttachAvailabilityChangeEvent:135: called
[  93.118105] HW.GPU <Info> video_core/texture_cache/texture_cache.h:QueueAsyncDecode:1201: Queuing async texture decode
[  93.350232] HW.GPU <Info> video_core/texture_cache/texture_cache.h:QueueAsyncDecode:1201: Queuing async texture decode
[  96.717686] Render.Vulkan <Critical> video_core\vulkan_common\vulkan_device.cpp:ReportLoss:538: Device loss occurred!

**Comment by Squall-Leonhart** [comment]
the critical is file system related and i believe it was introduced when liam added the cache storage support needed for totk to boot and read/write the fuse icons.

Turn off Async Texture decoding.

**Comment by MrCatters** [comment]
> the critical is file system related and i believe it was introduced when liam added the cache storage support needed for totk to boot and read/write the fuse icons.
> 
> Turn off Async Texture decoding.

I have been playing for around 10 minutes with this setting turned off. No crashes at all on my end now.

**Comment by Jaitsu** [comment]
FWIW, the "Assertion failed!" in fsp_srv.cpp is present even when the game starts successfully, and seems to have no negative effect that I can discern - TotK still loads/runs/saves correctly even with it present.

**Comment by UmbraAtrox** [comment]
I think i have the same issue 
[yuzu_log.txt](https://github.com/yuzu-emu/yuzu/files/11555823/yuzu_log.txt)

i5-12600k, 32gb ddr4, A750

**Comment by doggo404** [comment]
[yuzu_log.txt](https://github.com/yuzu-emu/yuzu/files/11568144/yuzu_log.txt)
I have the same issue.

Windows 11 22H2 insider build
Ryzen 5 2600
Geforce GTX 1650Super 4gb Vram (gigabyte windforce brand)
16gb ddr4 2933

**Comment by doggo404** [comment]
I got the game running, it works without crashes so far!

- (Nvidia only), download https://developer.nvidia.com/downloads/vulkan-beta-53204-windows
- Make sure to disable afterburner from starting with windows, revert any overclocks force quit.
- Open driver install, click agree
- Check  the "Custom (Advanced)" option, press next
- Check "Preform a Clean Installation" option, continue as normal with install, restart when asked.
- open yuzu...
- Profit!

**Comment by Infinality** [comment]
> I got the game running, it works without crashes so far!
> 
> * (Nvidia only), download https://developer.nvidia.com/downloads/vulkan-beta-53204-windows
> * **Make sure to disable afterburner from starting with windows, revert any overclocks, force quit.**
> * Open driver install, click agree
> * Check  the "Custom (Advanced)" option, press next
> * Check "Preform a Clean Installation" option, continue as normal with install, restart when asked.
> * open yuzu...
> * Profit!

Sadly the beta driver didn't help matters on my end at all.  Do you think "Perform a Clean Installation" was the key here?

**Comment by doggo404** [comment]
> Sadly the beta driver didn't help matters on my end at all. Do you think "Perform a Clean Installation" was the key here?

Im honestly sure afterburner is the issue.
i had an issue with cemu where botw would crash and literally not having any overclock, having no fps monitor and not having afterburner running would fix it completely.

**Comment by Infinality** [comment]
> Its definitely worth trying with a clean installation though!

I don't have Afterburner installed, so at least in my case it's not that.  I'll try a clean install and see if it helps.  My gut says this is something very specific to Yuzu, as Vulkan works not only on other programs, but even on certain games within Yuzu itself.  BOTW and TOTK however refuse to get past the launch screen.  What's strange is that BOTW used to work with Vulkan.

**Comment by doggo404** [comment]
> EDIT: Crashes still, but I have **2** "Device Loss Occurred" lines now. XD Not the profit I was hoping for! 

Thats such a shame, probably a yuzu issue so.

Some other troubleshooting steps you could do is to check if rivatuner statistics server is installed, if so update it.
You could also try https://github.com/13xforever/vkdiag . Yuzu themselves recommend it: 

>A recurring problem users face when running modern emulators that utilize Vulkan is the lack of a way to get helpful feedback if the program crashes. Vulkan is very picky. Any wrong step in the middle and it will refuse to start, and any Vulkan layer program can cause this. Anything from outdated screen recorders, to mod managers, performance overlays, and even the disaster code motherboard vendors call RGB software. Of particular nasty interest is the Epic Game Store’s overlay, which is reported to cause severe performance issues.

>To some extent, yuzu can detect these problems and switch to OpenGL to avoid them, but that doesn’t tell the user where the problem is. We recently found out about a cute little program the [RPCS3](https://github.com/RPCS3/rpcs3) community has been using for some time to diagnose these problems, [VkDiag](https://github.com/13xforever/vkdiag). You can find its latest version [here](https://github.com/13xforever/vkdiag/releases).

>What this program does is list all relevant components and layers affecting Vulkan rendering and warn about potential problems in any of them.

>We have started using it in our support channels with amazing success, and we recommend self-serving users to keep it in their toolkit. You never know when that OBS install you forgot to update can cause you problems, or when that RGB software decides to go rogue.

Check here: https://yuzu-emu.org/entry/yuzu-progress-report-apr-2023/

**Comment by Infinality** [comment]
> > EDIT: Crashes still, but I have **2** "Device Loss Occurred" lines now. XD Not the profit I was hoping for!
> 
> Thats such a shame, probably a yuzu issue so.
> 
> Some other troubleshooting steps you could do is to check if rivatuner statistics server is installed, if so update it. You could also try https://github.com/13xforever/vkdiag . Yuzu themselves recommend it:
> 
> > A recurring problem users face when running modern emulators that utilize Vulkan is the lack of a way to get helpful feedback if the program crashes. Vulkan is very picky. Any wrong step in the middle and it will refuse to start, and any Vulkan layer program can cause this. Anything from outdated screen recorders, to mod managers, performance overlays, and even the disaster code motherboard vendors call RGB software. Of particular nasty interest is the Epic Game Store’s overlay, which is reported to cause severe performance issues.
> 
> > To some extent, yuzu can detect these problems and switch to OpenGL to avoid them, but that doesn’t tell the user where the problem is. We recently found out about a cute little program the [RPCS3](https://github.com/RPCS3/rpcs3) community has been using for some time to diagnose these problems, [VkDiag](https://github.com/13xforever/vkdiag). You can find its latest version [here](https://github.com/13xforever/vkdiag/releases).

Holy crap!  This is a godsend.  It works now!   The diagnostic identified a couple things I had installed (but weren't even running).  I removed them, and BAM, it works now!  What an insidious issue.

THANK YOU!!!

**Comment by doggo404** [comment]
> Holy crap! This is a godsend. It works now! The diagnostic identified a couple things I had installed (but weren't even running). I removed them, and BAM, it works now! What an insidious issue.
> 
> THANK YOU!!!

I am glad it worked out for you! :D

**Comment by jdhau** [comment]
Running 3612 and so far, so good... Load the game 2x with no isues and no device losses a few minutes in. Nice work devs!

**Comment by jdhau** [comment]
3612 seems to have fixed this. Performance seems very poor (worse than Ryujinx), but that's a separate issue.

**Comment by Jaitsu** [comment]
@jdhau I still get the device loss on startup, unfortunately... but again, I'm convinced there are two different causes at work here. I've only once ever had the device loss occur _during gameplay_ (a minute after a fast travel), but every other time, it's occurred at startup, between the "Launching title..." 

I no longer think my "run nvidia control panel" fix above actually _fixes_ anything, but it _does_ seem to be the absolute deciding factor in whether a startup crash manifests as a permanent black-screen hang (waiting up to 5 minutes) until I kill the process, or a "Device loss occurred!" error and immediate crash.

_With_ something else other than Yuzu running on the dGPU in my hybrid-graphics laptop, I get 50/50 successful starts and black screens, _never_ device losses. _Without_ something else on the dGPU, I get 50/50 successful starts and device losses, _never_ black screen hangs. This has been repeatable dozens of times, and most interestingly of all, if I then close the other program while yuzu is sitting on the black screen, _then_ it crashes with a device loss less than a minute later. All mods removed for all tests.

I have logs of all 4 scenarios (successful start, black screen, device loss, black screen -> long delay, then close other program -> device loss):
[yuzu_log_longdelay.txt](https://github.com/yuzu-emu/yuzu/files/11570245/yuzu_log_longdelay.txt)
[yuzu_log_deviceloss.txt](https://github.com/yuzu-emu/yuzu/files/11570246/yuzu_log_deviceloss.txt)
[yuzu_log_permablackscreen.txt](https://github.com/yuzu-emu/yuzu/files/11570247/yuzu_log_permablackscreen.txt)
[yuzu_log_successfulstart.txt](https://github.com/yuzu-emu/yuzu/files/11570248/yuzu_log_successfulstart.txt)

**Comment by jdhau** [comment]
Sorry, have t reopen this as I just got a device loss when fast traveling :( Big improvement but not here yet.

**Comment by jdhau** [comment]
It looks like switching to BC3 texture compression removes my crashes. I can get the game stable now, as long as I don't muck around with other HW accelerated apps.

**Comment by jdhau** [comment]
Been playing around with this for a while and the game is definitely still not stable yet. I get some odd stuttering, and relatively frequent crashes on going in and out of menus.

**Comment by Jaitsu** [comment]
@MrCatters 
> This sounds like an issue caused by another program interfering with Yuzu. I would take a look at what [doggo404](https://github.com/doggo404) has posted earlier [here ](https://github.com/yuzu-emu/yuzu/issues/10290#issuecomment-1563442046) if you have not already.

No RTSS installed; when I say "the other program" I mean "literally any other program that keeps the dedicated GPU awake"; I use the Nvidia Control Panel as something quick to start up, but the same happens when running another lightweight game (that still uses the GPU), GPU utility, etc - as soon as yuzu is the only thing in the Nvidia GPU Activity window, **during that hang,** it crashes ~30 seconds later.

VkDiag.exe reports no issues other than me being on an insider Windows 11 build (beta ring - fairly stable, and no other game/software has any abnormalities I'm aware of) and wanting me to update my Intel (integrated) GPU drivers because they're ~2 months old, but that's the latest version available.
![image](https://github.com/yuzu-emu/yuzu/assets/463807/93946bce-3fbb-4d64-ae13-b462c551bba6)
